olive/workflows/run/config.py

```diff
@@ -354,36 +378,14 @@ def _resolve_config_str(v, values, alias, component_name):
 
 
 def _resolve_system(v, values, system_alias):
-    v = _resolve_config_str(v, values, system_alias, component_name="systems")
-    if v.get(system_alias):
-        v[system_alias] = validate_config(v[system_alias], SystemConfig)
-        if v[system_alias].type == "AzureML":
-            if not values["azureml_client"]:
-                raise ValueError("AzureML client config is required for AzureML system")
-            v[system_alias].config.azureml_client_config = values["azureml_client"]
-    return v
+    return _resolve_config_str(v, values, system_alias, component_name="systems")
 
 
 def _resolve_data_config(v, values, data_config_alias):
-    if not isinstance(v, dict):
-        # if not a dict, return the original value
-        return v
-
-    # get name of sub component
-    sub_component = v.get(data_config_alias)
-    if not isinstance(sub_component, str):
-        return v
-
-    # resolve data_config_alias to data config
-    components = values.get("data_configs") or []
-    component_map = {cmp.name: cmp for cmp in components}
-
-    # resolve sub component name to component config
-    if sub_component not in component_map:
-        raise ValueError(f"{data_config_alias} {sub_component} not found in {components}")
-
-    v[data_config_alias] = component_map[sub_component]
-    return v
+    # it is okay to insert a new member into values
+    # doesn't affect the original values
+    values["data_configs_dict"] = {dc.name: dc for dc in values.get("data_configs") or []}
```

Comment:
Though this doesn't affect the functionality, this value will be added to the parsed run config obj.

Reply:
When I checked last time, I didn't see this behavior but you are right, it shows up in the object. I will pass this directly as {"data_configs_dict": ...} into the next call

Reply:
updated